Content.Server/Disease/DiseaseZombieSystem.cs

```diff
@@ -0,0 +1,196 @@
+        [Dependency] private readonly DamageableSystem _damageable = default!;
+        [Dependency] private readonly BloodstreamSystem _bloodstream = default!;
+        [Dependency] private readonly MovementSpeedModifierSystem _movementSpeedModifier = default!;
+        [Dependency] private readonly SharedHandsSystem _sharedHands = default!;
+        [Dependency] private readonly SharedHumanoidAppearanceSystem _sharedHumanoidAppearance = default!;
+        [Dependency] private readonly IRobustRandom _robustRandom = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<DiseaseZombieComponent, ComponentInit>(OnComponentInit);
+            SubscribeLocalEvent<DiseaseZombieComponent, MeleeHitEvent>(OnMeleeHit);
+            SubscribeLocalEvent<DiseaseZombieComponent, RefreshMovementSpeedModifiersEvent>(OnRefreshMovementSpeedModifiers);
+        }
+        /// <remarks>
+        /// I would imagine that if this component got assigned to something other than a mob, it would throw hella errors.
+        /// </remarks>
+        private void OnComponentInit(EntityUid uid, DiseaseZombieComponent component, ComponentInit args)
+        {
+            if (!component.ApplyZombieTraits || !HasComp<MobStateComponent>(uid))
+                return;
+
+            if (HasComp<DiseaseCarrierComponent>(uid))
+                EntityManager.RemoveComponent<DiseaseCarrierComponent>(uid);
+            if (HasComp<RespiratorComponent>(uid))
+                EntityManager.RemoveComponent<RespiratorComponent>(uid);
+            if (HasComp<BarotraumaComponent>(uid))
+                EntityManager.RemoveComponent<BarotraumaComponent>(uid);
+            if (HasComp<HungerComponent>(uid))
+                EntityManager.RemoveComponent<HungerComponent>(uid);
+            if (HasComp<ThirstComponent>(uid))
+                EntityManager.RemoveComponent<ThirstComponent>(uid);
+
+            EntityManager.EnsureComponent<BloodstreamComponent>(uid, out var bloodstream); //zoms need bloodstream anyway for healing
+            _bloodstream.SetBloodLossThreshold(uid, 0f, bloodstream);
+            _bloodstream.TryModifyBleedAmount(uid, -bloodstream.BleedAmount, bloodstream);
+            _movementSpeedModifier.RefreshMovementSpeedModifiers(uid);
+            EntityManager.EnsureComponent<ReplacementAccentComponent>(uid).Accent = "zombie";
+
+            if (TryComp<DamageableComponent>(uid, out var comp))
+            {
+                _damageable.SetDamageModifierSetId(uid, "Zombie", comp);
+                _damageable.SetAllDamage(comp, 0);
+            }
+
+            // circumvents the 20 damage after being converted.
+            // idk a better way of doing this within the meleeHitEvent
+            var healingSolution = new Solution();
+            healingSolution.AddReagent("Bicaridine", 5.00);
+            _bloodstream.TryAddToChemicals(uid, healingSolution);  
```

Comment:
grab the damagablecomponent and directly remove 20 damage. Don't do funnies with chemistry.

Reply:
The issue is that the damage is applied after the component is initialized, making it so the damage will still be taken. Is there another way to get around that?